build_scripts/tests_runner.py

```diff
@@ -0,0 +1,114 @@
+import os
+import sys
+import logging
+import pathlib
+import argparse
+
+sys.path.append(str(pathlib.Path(__file__).resolve().parents[1]))
+from build_scripts.common_runner import ConfigGenerator
+from common.helper import TestStage, ErrorCode
+from common.logger_conf import configure_logger
+from common.manifest_manager import Manifest
+
+
+class TestsRunnerException(Exception):
+    pass
+
+
+class ArtifactsNotFoundException(TestsRunnerException):
+    pass
+
+
+class TestScenarioNotFoundException(TestsRunnerException):
+    pass
+
+
+class TestRunner(ConfigGenerator):
+    def __init__(self, artifacts, root_dir, current_stage):
+        self._artifacts_dir = None
+        self._manifest = None
+
+        if artifacts.exists():
+            if artifacts.is_file():
+                self._artifacts_dir = artifacts.parent
+                self._manifest = Manifest(artifacts)
+            else:
+                self._artifacts_dir = artifacts
+                self._manifest = Manifest(artifacts / 'manifest.yml')
```

Comment:
it should be checked that the manifest file exits
probably the constructor of Manifest manager should be imroved